### PR TITLE
Default env devx 2489

### PR DESCRIPTION
**Pull request by sarveshdream11**
Details of the PR: https://dream11.atlassian.net/browse/DEVX-2489

Automation job link: https://cistag-internal.dream11.com/job/Devx-Automation/847/console

### CODE REVIEW

File: api/configuration/configuration.go [hunk 1]
```diff
@@ -12,4 +12,5 @@ type Configuration struct {
 	Keys         SecretKeys
 	AccessToken  string `yaml:"access_token,omitempty" json:"access_token,omitempty"`
 	RefreshToken string `yaml:"refresh_token,omitempty" json:"refresh_token,omitempty"`
+	OdinEnvName  string `yaml:"odinEnvName,omitempty" json:"odinEnvName:,omitempty"`
```
Comment (akshaypatidar1999) [inline api/configuration/configuration.go:15]: Rename this to `EnvName`. No need to prefix odin

File: internal/backend/env.go [hunk 2]
```diff
@@ -153,3 +158,36 @@ func (e *Env) EnvTypes() (envResp.EnvTypesResponse, error) {
 
 	return envResponse, err
 }
+
+func (e *Env) SetEnv(envName string) error {
+	dirname, err := os.UserHomeDir()
+	if err != nil {
+		return err
+	}
+	data, err := file.Read(dirname + "/.odin/config")
```
Comment (surajgour-d11) [inline internal/backend/env.go:167]: odin config file can be found using `app.WorkDir.Location`, `app.WorkDir.ConfigFile`.
Refer to `commands/configure.go`

File: internal/backend/env.go [hunk 3]
```diff
@@ -153,3 +158,36 @@ func (e *Env) EnvTypes() (envResp.EnvTypesResponse, error) {
...
+	}
+	result := ""
+	r, _ := regexp.Compile(`(?:odinEnvName: [a-zA-Z]+-\w+)`)
+	match := r.FindString(string(data))
+	if match != "" {
+		result = strings.Replace(string(data), match, fmt.Sprintf("odinEnvName: %s", envName), 1)
+	} else {
+		result = string(data) + fmt.Sprintf("odinEnvName: %s\n", envName)
+	}
+	err = file.Write(dirname+"/.odin/config", result, 0755)
```
Comment (surajgour-d11) [inline internal/backend/env.go:179]: remove hardcoded directory path

File: pkg/utils/utils.go [hunk 4]
```diff
@@ -39,3 +44,33 @@ func Contains(s []string, str string) bool {
...
+		result = string(data) + fmt.Sprintf("envName: %s\n", envName)
+	}
+	err = file.Write(configPath, result, 0755)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+func FetchSetEnv(envName string) string {
```
Comment (akshaypatidar1999) [inline pkg/utils/utils.go:69]: can we rename this to `FetchEnv`

**Review by akshaypatidar1999 — APPROVED** [review]

File: pkg/utils/utils.go [hunk 5]
```diff
@@ -39,3 +44,33 @@ func Contains(s []string, str string) bool {
...
+		result = string(data) + fmt.Sprintf("envName: %s\n", envName)
+	}
+	err = file.Write(configPath, result, 0755)
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+func FetchEnv(envName string) string {
```
Comment (BalkishanNanavath) [inline pkg/utils/utils.go:69]: if we are fetching the env then why would we provide envName as the argument to the function?

File: internal/command/commands/serviceset.go [hunk 6]
```diff
@@ -136,6 +137,7 @@ func (s *ServiceSet) Run(args []string) int {
 	}
 
 	if s.Deploy {
+		*envName = utils.FetchEnv(*envName)
```
Comment (BalkishanNanavath) [inline internal/command/commands/serviceset.go:140]: we can directly use the envName right? the FetchEnv method is not clear

File: pkg/utils/utils.go [hunk 7]
```diff
@@ -39,3 +44,33 @@ func Contains(s []string, str string) bool {
 	}
 	return false
 }
+
+func SetEnv(envName string) error {
```
Comment (BalkishanNanavath) [inline pkg/utils/utils.go:48]: we can split this function into 2 functions. 1 to read/write data into a file and the other to look for a regex in string.

**Review by surajgour-d11 — APPROVED** [review]

File: internal/command/commands/env.go [hunk 8]
```diff
@@ -57,6 +59,9 @@ func (e *Env) Run(args []string) int {
 	}
 
 	if e.Status {
+		if *name == "" {
+			*name = utils.FetchKey("EnvName")
```
Comment (BalkishanNanavath) [inline internal/command/commands/env.go:63]: can we create a constant for the key "EnvName" and use the constant everywhere? this way if the key changes then you would need to just change the constant value.

**Pull request merged**